docs/quantum_gate.rst

```diff
@@ -1,4 +1,366 @@
+.. _quantum_gates:
+
 Quantum Gates
 =============
 
-contains gate details
\ No newline at end of file
+Gates in OpenQL are the constructs that refer to operations to be executed somehow on the computing platform.
+
+A gate refers to an operation and to zero or more operands.
+
+Gates are organized in circuits as vectors of gates, i.e. linear sequences of gates.
+A circuit defines the operation of a kernel.
+And a program consists of multiple kernels.
+
+Gates can be subdivided in several kinds.
+This is useful in the description of the passes below.
+
+
+First, gates can be subdivided according to where their execution has effect:
+
+- quantum gates; these gates execute in the quantum computing hardware; these gates have at least one qubit as (implicit or explicit) operand; these gates can have classical registers as operand as well and may rely on some execution capability in classical hardware
+
+- classical gates; these gates execute in classical hardware only; these don't have any qubit as operand, only zero or more classical registers
+
+- directives; these gates execute neither in quantum nor in classical hardware; these look like gates but don't influence execution, e.g. the display gate
+
+
+Quantum gates can be subdivided in several ways from the number of operands they take; this becomes relevant when gates are mapped on the quantum computing platform:
+
+- one-qubit gates; quantum gates operating on one qubit
+
+- two-qubit gates; quantum gates operating on two qubits;
+  these are the main objective in mapping since two-qubit gates require their qubit operands to be connected in the hardware
+
+- multi-qubit gates; quantum gates operating (implicitly or explicitly) on more than two qubits;
+  these must be decomposed to one-qubit and two-qubit gates because more-qubit primitive gates are not supported by the quantum platform
```

Comment:
"because more-qubit primitive gates are not supported by the quantum platform" 
Does this mean that OpenQL is fundamentally incompatible with hardware platforms that specify multi-qubit primitives? 

We have a multiplexed readout operation that is fundamentally multi-qubit.

Reply:
OpenQL can handle gates with any number of qubits.
See measure, barrier, wait, etc.

Updated the documentation to make clear that the examples given reflect the current mapping pass that targets CC-Light